src/runs.rs

```diff
@@ -0,0 +1,341 @@
+
+        Some(pid)
+    }
+
+    pub fn read_infos(&self) -> Option<DaemonRunInfos> {
+        if !self.infos_file_path().exists() {
+            return None;
+        }
+
+        let infos_json = std::fs::read_to_string(self.infos_file_path()).unwrap();
+        let infos: DaemonRunInfos = serde_json::from_str(&infos_json).unwrap();
+
+        Some(infos)
+    }
+
+    pub fn read_stdout(&self) -> String {
+        std::fs::read_to_string(self.stdout_path()).unwrap()
+    }
+
+    pub fn read_stderr(&self) -> String {
+        std::fs::read_to_string(self.stderr_path()).unwrap()
+    }
+
+    pub fn start(&self, task: Vec<String>) -> miette::Result<()> {
+        #[cfg(unix)]
+        {
+            // Create stdout and stderr files
+            let stdout = std::fs::File::create(self.stdout_path()).unwrap();
+            let stderr = std::fs::File::create(self.stderr_path()).unwrap();
+
+            // Create and save the infos file
+            let infos = DaemonRunInfos {
+                name: self.name.clone(),
+                task,
+                start_date: Local::now(),
+            };
+            let infos_json = serde_json::to_string_pretty(&infos).unwrap();
+            std::fs::write(self.infos_file_path(), infos_json).unwrap();
+
+            // Create the daemon
+            let daemonize = Daemonize::new()
+                .pid_file(self.pid_file_path())
+                .stdout(stdout)
+                .stderr(stderr)
+                .umask(0o027) // Set umask, `0o027` by default.
+                .chown_pid_file(true)
+                .working_directory(self.working_dir.clone());
+
+            // Start the daemon
+            daemonize.start().into_diagnostic()
+        }
+        #[cfg(not(unix))]
+        {
+            let _ = task;
+            miette::bail!("The `start` command is only available on Unix systems.");
+        }
+    }
+
+    pub fn clear(&self) -> miette::Result<()> {
+        // check if the run is running
+        if self.is_running() {
+            miette::bail!("The run is still running. You can call `pixi runs kill` to kill it.");
+        }
+
+        // delete pid, infos, stdout and stderr files
+        std::fs::remove_file(self.pid_file_path()).expect("Failed to remove pid file");
+        std::fs::remove_file(self.infos_file_path()).expect("Failed to remove infos file");
+        std::fs::remove_file(self.stdout_path()).expect("Failed to remove stdout file");
+        std::fs::remove_file(self.stderr_path()).expect("Failed to remove stderr file");
+
+        Ok(())
+    }
+
+    pub fn state(&self) -> miette::Result<DaemonRunState> {
+        let pid = match self.read_pid() {
+            Some(pid) => pid,
+            None => miette::bail!("No pid file with name '{}' found.", self.name),
+        };
+        let stdout_length = std::fs::read_to_string(self.stdout_path()).unwrap().len();
+        let stderr_length = std::fs::read_to_string(self.stderr_path()).unwrap().len();
```

Comment:
I think these should be errors too, a malicious user could have just removed them. We don't want to crash in that case.